client/hosting/hosting-features/components/hosting-features.tsx

```diff
@@ -58,34 +58,35 @@ const HostingFeatures = () => {
 	);
 	const hasEnTranslation = useHasEnTranslation();
 
-	const transfer = useSelector( ( state ) => getAtomicTransfer( state, siteId ) );
-	const isTransferInProgress = [
-		transferStates.PENDING,
-		transferStates.ACTIVE,
-		transferStates.PROVISIONED,
-	].includes( transfer.status );
+	const { data: siteTransferData, refetch: refetchSiteTransferData } = useSiteTransferStatusQuery(
+		siteId || undefined
+	);
+	// We have some bug at the backend, when the transfer is complited, but isSiteAtomic is still false, so we need such extra condition
+	const isTransferInProgress =
+		siteTransferData?.isTransferring || siteTransferData?.status === transferStates.COMPLETED;
 
 	useEffect( () => {
 		if ( ! siteId ) {
 			return;
 		}
 
-		if ( transfer.status === transferStates.COMPLETED ) {
-			window.location.href = `/overview/${ siteSlug }`;
-			return;
-		}
-
-		dispatch( fetchAtomicTransfer( siteId ) as unknown as AnyAction );
-		// A user can keep one tab open and start transfer in the second tab
-		// se we always run interval to check the status, to avoid case with rendering "Activate" button on old tabs of a browser
 		const interval = setInterval( () => {
-			dispatch( fetchAtomicTransfer( siteId ) as unknown as AnyAction );
-		}, 10000 );
+			if ( siteTransferData?.status !== transferStates.COMPLETED ) {
+				refetchSiteTransferData();
+			} else {
+				clearInterval( interval );
+				dispatch( fetchAtomicTransfer( siteId ) as unknown as AnyAction );
+			}
+		}, 3000 );
 
-		return () => {
-			clearInterval( interval );
-		};
-	}, [ siteSlug, siteId, transfer.status, dispatch ] );
+		return () => clearInterval( interval );
+	}, [ siteId, siteTransferData?.status, refetchSiteTransferData, dispatch ] );
+
+	useEffect( () => {
+		if ( isSiteAtomic && ! isPlanExpired ) {
+			page.replace( redirectUrl.current );
+		}
```

Comment:
```suggestion
		if ( isSiteAtomic && ! isPlanExpired ) {
			setTimeout( () => {
				page.replace( redirectUrl.current );
			}, 3000 );
		}
```
With this change, I had three consecutive successful test attempts, so @chad1008's [idea](https://github.com/Automattic/wp-calypso/pull/93488#issuecomment-2289064672) about the redirect happening too early could definitely be pointing in the right direction.

That's not saying I think we should land this approach, but it might be helpful for debugging nonetheless.

Reply:
I tested SetTimouts in different places - it should not be the reason, most likely it was coincidence that it was successful for you. I tested even with 15-30 secs delay.
Also, if at the end of the day we decide to proceed with some timeouts - we need to remember, they redirect was used before and it should word immediately for the old cases. Thta's why any timeout as a workaround could break something else and make maintenance more difficult, that's why I decided not to consider it at all.

Reply:
😔